content/src/logic/time-range.ts

```diff
@@ -104,3 +104,34 @@ export function divideTimeInYearsMonthsWeeksAndDays(timeRange: TimeRange): TimeR
     }
   }
 }
+
+export function joinOverlappedTimeRanges(timeRanges: TimeRange[]): TimeRange[] {
```

Comment:
would be possible to add a comment here explaining the expected result of the function?

Reply:
I also want to point out that there is a very good node library for handling all time stuff, in this case intervals: https://date-fns.org/v2.30.0/docs/Interval, not saying you should use it or that it has everything you need, but I remember the utilities were very expressive and the code using the library was very easy to follow

Reply:
Will add a comment here and take a look that that lib 👍